### ISSUE TITLE
Media Player doesn't work with Cozy

### ISSUE THREAD

**Issue by steve5289** [azymohliad/watchmate#26]
Not sure if this is a Cozy, Infinitime, or watchmate issue.
## What I see
When cozy is running(whether it is playing anything makes no difference): 
   * watchmate will correctly show Cozy as the "Media Player"
   * The pinetime just shows the last music I played in lollypop, and buttons do nothing.
   * If I open lollypop back up to play music, everything works fine without any fiddling.

## Things Attempted
 1. I tried playing a 1 minute audiobook chapter just in case it was an overflow issue(most audiobook chapters have far greater run lengths than music). No effect.
 2. Resetting the pinetime (holding the button down until it resets). No effect

## Software Versions
* watchmate(flatpak): 0.4.5
* Cozy: 1.2.1
* Infinitime: 1.12.0

Cozy: https://github.com/geigi/cozy

**Comment by azymohliad** [comment]
Thanks! It is a WatchMate issue, I reproduced and found the rootcause (Cozy doesn't have `LoopStatus` property, and WatchMate hangs media player control session waiting for it, even though it's optional by the standard). Looking for the best fix now

**Comment by azymohliad** [comment]
It actually turns out to be a Cozy issue, missing property requests shouldn't hang the client (details are [here](https://github.com/geigi/cozy/issues/777)). It can be worked around in WatchMate, but not without tradeoffs. The easiest thing WatchMate could do is to not read loop and shuffle statuses at all (Infinitime doesn't display them currently anyway), but Infinitime has API for setting them, so I'd prefer to preserve the full implementation, in case Infinitime starts displaying those at some point.

I might contribute the fix to Cozy a bit later if that would help to resolve it faster, it is pretty simple.

**Comment by steve5289** [comment]
Let me see if I can figure it out. I'm trying to learn how to develop using gtk and you've pointed me in a direction that I think I can dig into. If it's ok I may ask some questions.

**Comment by azymohliad** [comment]
Well, there's no GTK really involved in the fix (unless you mean the stack beneath it too, such as GIO D-Bus helpers). I think what I described in my Cozy bug report would be the whole fix. I'd wait for Cozy maintainers to respond there, just to be sure they agree with it, and maybe they have a specific preferred way to implement it (for example, raise an exception in `MPRIS.Get()`, or keep `MPRIS` class intact, and rather process its return value in `Server.on_method_call`. Although maybe they would prefer a PR already, idk. I'm of course happy to answer any questions as much as my knowledge allows

**Comment by steve5289** [comment]
I've mostly just written daemons, and command line based stuff so to me it's the audio/visual stuff that I'm trying to learn how it all works enough to be competent at writing/maintaining it. 

Ok, I'll hold off on creating something for this issue, yeah your bug report basically spells out the exact proposed solution. Your explanation of this issue has helped me understand another bug in cozy that I'll try to track down though.

**Comment by azymohliad** [comment]
Hey, just FYI I submitted a fix [here](https://github.com/geigi/cozy/pull/779), in case you'd want to try it. I guess Cozy dev isn't actively working on the app atm, so I figured it would be more useful to submit a patch.

I also noticed that turning the volume down to 0 with PineTime/WatchMate (or any other MPRIS client) would break Cozy media player control again, but that might be a GStreamer issue (more details [here](https://github.com/geigi/cozy/issues/780)).
